Add scalar tensor system and tags

guilara opened the pull request:
## Proposed changes

Add new namespace  and tags/aliases for the combined system of CurvedScalarWave and generalized harmonic.

<!--
Add new namespace  and tags/aliases for the combined system of CurvedScalarWave and generalized harmonic.
-->

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

guilara committed:
Add scalar tensor system and tags

nilsvu commented on src/Evolution/Systems/ScalarTensor/System.hpp:
```diff
@@ -0,0 +1,44 @@
+
+/*!
+ * \ingroup EvolutionSystemsGroup
+ * \brief Items related to evolving the first-order scalar tensor system.
```
Please add some docs here :) Add the equations and/or link to the GH and CSW systems.

guilara replied:
Added an expanded description below.

nilsvu commented on src/Evolution/Systems/ScalarTensor/Tags.hpp:
```diff
@@ -0,0 +1,22 @@
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace ScalarTensor::Tags {
```
Add docstring so this shows up in the dox

nilsvu commented on src/Evolution/Systems/ScalarTensor/Tags.hpp:
```diff
@@ -0,0 +1,22 @@
+ * \brief Represents the trace-reversed stress-energy tensor of the scalar
+ * field.
+ */
+struct TraceReversedStressEnergy : db::SimpleTag {
```
Please add some docs in the `ScalarTensor` namespace to clarify where you need this

nilsvu commented on src/Evolution/Systems/ScalarTensor/Tags.hpp:
```diff
@@ -0,0 +1,25 @@
+ * \brief Represents the trace-reversed stress-energy tensor of the scalar
+ * field.
+ */
+struct TraceReversedStressEnergy : db::SimpleTag {
```
Please template on `typename DataType, size_t Dim, typename Fr = Frame::Inertial>` for consistency with other tensor tags.

guilara replied:
Added template

nilsvu commented on src/Evolution/Systems/ScalarTensor/Tags.hpp:
```diff
@@ -0,0 +1,26 @@
+ */
+template <typename DataType, size_t Dim, typename Fr = Frame::Inertial>
+struct TraceReversedStressEnergy : db::SimpleTag {
+  using type = tnsr::aa<DataType, Dim>;
```
`tnsr::aa<DataType, Dim, Fr>`

guilara replied:
Thanks Nils, I've made the change

nilsvu reviewed (changes requested):
Looks good, please squash this in.

nilsvu reviewed (dismissed):
Looks good. CI has a few complaints. Also, do you want to wait for the other reviews that you requested?

nilsdeppe reviewed (commented):
LGTM, I don't need to look at it again for it to get merged :)

guilara commented:
I updated the test. Hopefully it won't complain now.

I added @nikwit to make him aware of this PR. I think it his input will be useful later on.

nilsvu commented:
Just one CI complaint left

guilara commented:
I think the last complaint was due to cmake looking for the .hpp files in the CMakeLists.txt file. Previously I had only written the license in it as I didn't want to add those details now. In the end I've added more details to that file. I also added a couple of mostly empty files (StressEnergy.hpp/cpp) to pass at least one 'source' file to cmake and avoid more complaints.

nilsvu reviewed (approved):
The empty files aren't necessary (CMake would be fine without a `sources` list for example), but don't hurt either.

Want to wait for @nikwit before merging?

guilara commented:
I would say not to wait as this is just tags. But I'll ask for his review for the following
